metrics/mobile-app-metrics.md

```diff
@@ -0,0 +1,108 @@
+# Abstract
+
+This document contains the proposal for gathering metrics on system statistics for running Mobile Apps and making them available to developers.
+
+## Goals
+
+### Primary Goals
+
+- Provide mobile app developers a service that can receive overall runtime metrics (processor and memory data, authentication events, runtime versions, networking events, etc).
+
+- Provide developers an interface to query and visualize the gathered data in order to make informed decisions towards deployment, deprecation of backend services and other actions that depend on the state of an application's installed base.
+
+### Secondary Goals
+
+- Provide mobile app developers with an easy to use SDK to connect easily and securely to the metrics service
+
+### Non Goals
+
+- Identify individual devices or users in metrics events
+
+- Registering of user-defined metrics and logs
+
+## Terms
+
+For a list of terms used in this document, see [Glossary](./glossary.md).
+
+## Problem Description
+
+As Mobile Apps are developed, improved and iterated on, different versions of those Apps can be used on many different devices. System metrics data becomes invaluable for developers to observe the landscape of application installations and make sure further development and deployment of backend services will not affect users.
+
+## Proposed Solution
+
+A Metrics Service will be available from the Service Catalog, which can be provisioned by developers for use with their mobile applications.
+
+The extra components outlined in this proposal will be added to the existing components outlined in the [Metrics Endpoints and Auto Discovery Proposal](./mobile-app-metrics-service.md).
+
+The following diagram illustrates the components of the Metrics Service:
+
+![service diagram](./app-service-diagram.jpg)
+
+### Mobile App Client
+
+A native, platform-specific SDK for targetting the Metrics Service will be available to developers.
+
+Upon installation into a project and providing configuration for targetting the Metrics Service the SDK will automatically send general information about the general runtime for this application with a unique device ID, namely the `[SDK Version, clientID, timestamp]` tuple upon application initialization.
```

Comment:
@david-martin @wtrocki @pb82
Upon installation or after every app start? I think to answer "how many app starts happened in last N days with sdk version 1.2.3?", we need to send it every app start.

Reply:
Yes we need app start to get data about current usage. Not sure if it even makes sense to send a metrics event on install: it would make sense if we also recorded uninstall (to get the accurate numbers of installations). But as far as i know it's not possible for an App to know when it's uninstalled. So maybe something like `distinct(clientId)` for the total number of installations?